### PR TITLE
Charm v8

### PR DESCRIPTION
**Pull request by SimonRichardson**
This is a very mechanical change, nothing of note, other than some new
indirect go modules, that we should keep our eyes on.

Additionally, bump the charm repo version from v5 to v6 …

Now that we have a breaking change of charm, we should also state that
there might be consequences from charm v8 that also might impact the
charm repo. Moving that here forces us to reconcile that.

### CODE REVIEW

File: testing/mockstore.go [hunk 1]
```diff
@@ -13,12 +13,12 @@ import (
 	"strconv"
 	"strings"
 
-	"github.com/juju/charm/v7"
+	"github.com/juju/charm/v8"
 	"github.com/juju/loggo"
 	"github.com/juju/utils"
 	gc "gopkg.in/check.v1"
```
Comment (achilleasa) [inline testing/mockstore.go:20]: Redundant line?

File: testing/testcharm_test.go [hunk 2]
```diff
@@ -1,13 +1,13 @@
...
+package testing_test // import "github.com/juju/charmrepo/v6/testing"
 
 import (
-	"github.com/juju/charm/v7"
-	"github.com/juju/charm/v7/resource"
+	"github.com/juju/charm/v8"
+	"github.com/juju/charm/v8/resource"
 	jc "github.com/juju/testing/checkers"
 	"github.com/juju/testing/filetesting"
 	gc "gopkg.in/check.v1"
```
Comment (achilleasa) [inline testing/testcharm_test.go:9]: Redundant line?

File: csclient/params/params_test.go [hunk 3]
```diff
@@ -11,7 +11,7 @@ import (
 	gc "gopkg.in/check.v1"
 	"gopkg.in/macaroon-bakery.v2/httpbakery"
```
Comment (achilleasa) [inline csclient/params/params_test.go:13]: Do we need an empty line here?

**Comment by SimonRichardson** [comment]
$$merge$$

**Pull request merged**